src/addons/action-form.js

```diff
@@ -45,7 +51,13 @@ class ActionForm extends React.PureComponent {
   }
 
   submit(event: Event) {
-    let form = this.refs.form
+    let form = this.form
+
+    console.assert(
+      form,
+      'ActionForm submit was called before this component was mounted.'
+    )
```

Comment:
Added a basic assertion here for strict mode. This should never happen, but impossible states like to happen in the wild... Hopefully this makes debugging easier.